http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/LeoAuthProviderHelper.scala

```diff
@@ -1,95 +0,0 @@
-package org.broadinstitute.dsde.workbench.leonardo.auth
-
-import akka.actor.ActorSystem
-import akka.http.scaladsl.model.StatusCodes
-import com.typesafe.config.Config
-import com.typesafe.scalalogging.LazyLogging
-import java.util.UUID.randomUUID
-import net.ceedubs.ficus.Ficus._
-import org.broadinstitute.dsde.workbench.leonardo.model._
-import org.broadinstitute.dsde.workbench.leonardo.model.google.ClusterName
-import org.broadinstitute.dsde.workbench.model.{TraceId, UserInfo, WorkbenchEmail}
-import org.broadinstitute.dsde.workbench.model.google.GoogleProject
-import org.broadinstitute.dsde.workbench.util.FutureSupport
-
-import scala.concurrent.duration._
-import scala.concurrent.{ExecutionContext, Future, TimeoutException}
-import scala.util.control.NonFatal
-
-case class AuthProviderException(authProviderClassName: String,
-                                 isTimeout: Boolean = false,
-                                 traceId: TraceId)
-  extends LeoException(s"[$traceId] An authorization step ${if (isTimeout) "timed out." else "failed."}", StatusCodes.InternalServerError)
-
-/**
-  * Wraps a LeoAuthProvider and provides error handling so provider-thrown errors don't bubble up our app.
-  */
-object LeoAuthProviderHelper {
-  def apply(wrappedAuthProvider: LeoAuthProvider, config: Config, serviceAccountProvider: ServiceAccountProvider)(implicit system: ActorSystem): LeoAuthProviderHelper = {
-    new LeoAuthProviderHelper(wrappedAuthProvider, config, serviceAccountProvider)
-  }
-
-  def create(className: String, config: Config, serviceAccountProvider: ServiceAccountProvider)(implicit system: ActorSystem): LeoAuthProviderHelper = {
-    val authProvider = Class.forName(className)
-      .getConstructor(classOf[Config], classOf[ServiceAccountProvider])
-      .newInstance(config, serviceAccountProvider)
-      .asInstanceOf[LeoAuthProvider]
-
-    apply(authProvider, config, serviceAccountProvider)
-  }
-}
-
-class LeoAuthProviderHelper(wrappedAuthProvider: LeoAuthProvider, authConfig: Config, serviceAccountProvider: ServiceAccountProvider)(implicit system: ActorSystem)
-  extends LeoAuthProvider(authConfig, serviceAccountProvider) with FutureSupport with LazyLogging {
-
-  // Default timeout is specified in reference.conf
-  private lazy val providerTimeout = authConfig.as[FiniteDuration]("providerTimeout")
-  private implicit val scheduler = system.scheduler
-
-  private def safeCall[T](traceId: TraceId = TraceId(randomUUID))(future: => Future[T])(implicit executionContext: ExecutionContext): Future[T] = {
-    val exceptionHandler: PartialFunction[Throwable, Future[Nothing]] = {
-      case e: LeoException => Future.failed(e)
-      case te: TimeoutException =>
```

Comment:
It should be ok to remove this `Helper` layer... but it did 2 things:
1. Map internal Sam errors to a more  user-friendly `AuthProviderException`
2. Enforce a client side timeout for Sam calls (see `future.withTimeout` below). If Sam is slow or down I didn't want timeouts to propagate to the Leo API layer.

Ideally the `http4s` solution would handle these 2 things as well.

Reply:
timeout is enforced slightly different. After this PR, it'll be a max retry reached error. For translating errors to `AuthProviderException`, I kinda prefer it handled in route level and leo internals should worry much about how these exceptions are translated to API errors (something like https://github.com/DataBiosphere/welder/blob/master/server/src/main/scala/org/broadinstitute/dsp/workbench/welder/server/WelderApp.scala#L23)...Caller of some of these methods of auth provider might translate errors to other errors.